.github/dependabot.yml

```diff
@@ -0,0 +1,23 @@
+version: 2
+
+updates:
+  - package-ecosystem: github-actions
+    directory: /
+    schedule:
+      interval: weekly
+
+  - package-ecosystem: npm
+    allow:
+      - dependency-type: production
+    directory: /
+    schedule:
+      interval: weekly
+    versioning-strategy: widen
```

Comment:
Updates `dependencies` on a weekly schedule using the `widen` version strategy

> Widen the allowed version requirements to include both the new and old versions, when possible. Typically, this only increases the maximum allowed version requirement.

This should only open a PR when there is a new major version of a dependency. There is also configuration to add a [`cooldown`](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/dependabot-options-reference#cooldown-) to delay the PR by a configurable number of days.

I think this is optional, as major versions typically require code changes too..